presto-main/src/main/java/com/facebook/presto/operator/aggregation/state/DigestAndPercentileArrayStateSerializer.java

```diff
@@ -0,0 +1,81 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.operator.aggregation.state;
+
+import com.facebook.presto.spi.block.Block;
+import com.facebook.presto.spi.block.BlockBuilder;
+import com.facebook.presto.spi.type.Type;
+import com.google.common.collect.ImmutableList;
+import io.airlift.slice.DynamicSliceOutput;
+import io.airlift.slice.Slice;
+import io.airlift.slice.SliceInput;
+import io.airlift.stats.QuantileDigest;
+
+import java.util.List;
+
+import static com.facebook.presto.spi.type.VarcharType.VARCHAR;
+import static io.airlift.slice.SizeOf.SIZE_OF_DOUBLE;
+
+public class DigestAndPercentileArrayStateSerializer
+        implements AccumulatorStateSerializer<DigestAndPercentileArrayState>
+{
+    @Override
+    public Type getSerializedType()
+    {
+        return VARCHAR;
+    }
+
+    @Override
+    public void serialize(DigestAndPercentileArrayState state, BlockBuilder out)
+    {
+        if (state.getDigest() == null) {
+            out.appendNull();
+        }
+        else {
+            DynamicSliceOutput sliceOutput = new DynamicSliceOutput(state.getDigest().estimatedSerializedSizeInBytes() + SIZE_OF_DOUBLE);
```

Comment:
This is a list, so getSize is not necessarily an accurate measurement of how much space has been reserved. In practice we will be using an ImmutableList, which probably reflects the actual size better. What's the way to go here?